### PR TITLE
add ability to write secrets for amplify app

### LINKED ISSUE #31: Secret storage
### Describe the Feature

https://docs.aws.amazon.com/amplify/latest/userguide/environment-variables.html#environment-secrets

Amplify can access secrets during build if they're stored at a specific location, and they are encrypted using the default KMS key.

### Expected Behavior

It would be good to be able to write these secrets using this module.

### Use Case

^

### Describe Ideal Solution

^

### Alternatives Considered

_No response_

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by kevcube**
## what

Amplify can access secrets during build and runtime
This PR allows the 

## why

It may be useful depending on a user's secrets management pattern to write secrets here when creating the amplify app.

## references

https://docs.aws.amazon.com/amplify/latest/userguide/environment-variables.html#environment-secrets
fixes #31

### CODE REVIEW

**Comment by goruha** [comment]
@kevcube, why can't a secret be written externally based on the module's outputs after the Amplify app is created?

**Review by goruha — COMMENTED** [review]
@kevcube, why can't a secret be written externally based on the module's outputs after the Amplify app is created?

**Comment by kevcube** [comment]
> @kevcube, why can't a secret be written externally based on the module's outputs after the Amplify app is created?

It can be written externally, if someone wants to use remote-state or something to grab the amplify ID that's an option, but because amplify has the ability to access secrets stored at these predefined paths I thought it made sense to package in this  module because they can be considered a part of the amplify application stack

**Review by goruha — APPROVED** [review]

**Review by goruha — CHANGES_REQUESTED** [review]
@kevcube tests failed 
https://github.com/cloudposse/actions/actions/runs/10102221318/job/27937411342

File: main.tf [hunk 1]
```diff
@@ -127,3 +127,19 @@ resource "aws_amplify_webhook" "default" {
...
+
+  parameter_write = [for secret_key, secret_value in each.value.secrets : {
+    name        = format("/amplify/%s/%s/%s", one(aws_amplify_app.default[*].id), each.key, secret_key)
+    value       = secret_value
+    type        = "SecureString"
+    description = "AWS Amplify secret."
+  }]
+
+  context = module.this.context
+}
```
Comment (goruha) [inline main.tf:145]:
```suggestion
resource "aws_ssm_parameter" "default" {
  for_each    = local.environments
  name          = format("/amplify/%s/%s/%s", one(aws_amplify_app.default[*].id), each.key, secret_key)
  description = "AWS Amplify secret."
  type          = "SecureString"
  tier               = "Standard"
  value            = secret_value
  overwrite     = "false"
  data_type    = "text"
  
  tags = module.this.tags
}
```

**Review by goruha — CHANGES_REQUESTED** [review]

**Pull request closed without merging**